src/r4300/r4300_core.c

```diff
@@ -21,8 +21,89 @@
 
 #include "r4300_core.h"
 
+#include "cached_interp.h"
+#include "new_dynarec/new_dynarec.h"
+#include "r4300.h"
+
 void init_r4300(struct r4300_core* r4300)
 {
     init_mi(&r4300->mi);
 }
 
+int64_t* r4300_regs(void)
+{
+    return (int64_t*)reg;
+}
+
+int64_t* r4300_mult_hi(void)
+{
+    return (int64_t*)&hi;
+}
+
+int64_t* r4300_mult_lo(void)
+{
+    return (int64_t*)&lo;
+}
+
+unsigned int* r4300_llbit(void)
+{
+    return (unsigned int*)&llbit;
+}
+
+uint32_t* r4300_pc(void)
+{
+#ifdef NEW_DYNAREC
+    return (r4300emu == CORE_DYNAREC)
+        ? (uint32_t*)&pcaddr
+        : (uint32_t*)&PC->addr;
+#else
+    return (uint32_t*)&PC->addr;
+#endif
+}
+
+uint32_t* r4300_last_addr(void)
+{
+    return (uint32_t*)&last_addr;
+}
+
+unsigned int* r4300_next_interrupt(void)
+{
+    return &next_interupt;
+}
+
+unsigned int get_r4300_emumode(void)
+{
+    return r4300emu;
+}
+
+
+
+void invalidate_r4300_cached_code(uint32_t address, size_t size)
+{
+    if (r4300emu != CORE_PURE_INTERPRETER)
+    {
+#ifdef NEW_DYNAREC
+        invalidate_cached_code_new_dynarec(address, size);
+#else
+        invalidate_cached_code_hacktarux(address, size);
+#endif
```

Comment:
Does this correctly deal with the case where the user is using the Cached Interpreter but the New Dynarec is compiled in?

It seems there's a missing check for `r4300emu` beyond the `#ifdef`s.